### PR TITLE
feat(ios): Added ios support to existing project

### PR DESCRIPTION
**Pull request by warmachinesocial**
I have created the issue #754 and this is the PR however please let me know about the commit message I don't know if it's correct or not.

### CODE REVIEW

**Comment by Leptopoda** [comment]
The part in the brackets should resemble the backage you have changed.
At a quick glance looks like you should have use both app and neon, becoming something like:
`feat(app,neon): initial iOS support`

We also have sort of commit linting wich should have been activated by the `./tool/setup.sh` script if run by you.
running `dart run husky install` would install it without running the setup script (although I don't see a reason not to run the script). 
This will install a git hook to the local clone and whenever you commit the hook should be executed informing you whether you did it right (You mentioned to use a graphical git app. I don't know whether it supports this).

Finally please also signoff your commits. this can be done by using `git commit -s` (again might be different in a gui). The logs of the failed DCO check also talks you through that.

**Comment by Leptopoda** [comment]
I'll take a look at the PR in the coming days (I've got a old mac from a friend but haven't done anything to it yet).
I've never developed on/for Apple so I'll probably have to learrn a bit before approving the PR.

However I can immediately tell that this definitely relies on #447

File: tool/dev.sh [hunk 1]
```diff
@@ -4,5 +4,5 @@ cd "$(dirname "$0")/.."
 
 ./tool/build-dev-container.sh
 
-echo "Running development instance on http://localhost. To access it in an Android Emulator use http://10.0.2.2"
+echo "Running development instance on http://localhost. To access it in an Android Emulator use http://10.0.2.2 and on IOS emulator use http://localhost"
```
Comment (Leptopoda) [inline tool/dev.sh:7]: I think this is redundant as the message mentions `localhost` and only explicitly mentions `10.0.2.2` for an android emulator.

Please revert that

File: packages/neon/neon/lib/l10n/localizations_en.dart
Comment (Leptopoda) [inline packages/neon/neon/lib/l10n/localizations_en.dart:1]: Flutter automatically runs the gen-l10n command on hot restarts. Sadly this does not trigger a reformat.

Please ensure your files are propperly formatted by running `melos run format` or `fvm dart format --line-length 120 .` before commiting. (revert this and other unformatted files)

File: packages/app/test/widget_test.dart
Comment (Leptopoda) [inline packages/app/test/widget_test.dart:1]: This probably came with a `flutter create`.

Please remove this test as it's not needed for us.

File: packages/app/ios/Runner/Assets.xcassets/AppIcon.appiconset/Icon-App-1024x1024@1x.png
Comment (Leptopoda) [inline packages/app/ios/Runner/Assets.xcassets/AppIcon.appiconset/Icon-App-1024x1024@1x.png:1]: Wrong icon.

We currently have a custom script generating them but will probably change it in #447

File: packages/app/.metadata
Comment (Leptopoda) [inline packages/app/.metadata:1]: Ok it's odd that we didn't already have this file but I think it should also list android in there.

**Review by Leptopoda — CHANGES_REQUESTED** [review]
<s>Some initial feedback. If you have the time you can also think about adding iOS support to the separate neon apps (neon_files, neon_news and neon_notes).
We should also evaluate how we can best tackle the notifications support..</s>


Please understand that we are not going to merge this PR until we have a working iOS development setup and gathered some knowledge on how to do iOS flutter development. We are currently also involved in many other issues and short on time so please be patient.

File: packages/neon/neon/lib/src/platform/ios.dart [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+class IosNeonPlatform implements NeonPlatform {
+  const IosNeonPlatform();
+
+  @override
+  bool get canUseCamera => true;
+
+  @override
+  bool get canUsePushNotifications => false;
+
+//don't know what this means
```
Comment (Leptopoda) [inline packages/neon/neon/lib/src/platform/ios.dart:19]: please remove this comment.
This means whether the platform is supported by https://pub.dev/packages/quick_actions wich is true for iOS.
quick_actions are the actions you get when long pressing an app icon on the home screen (at least that's what it is on Android)

**Comment by provokateurin** [comment]
@nishp77 Can you make sure that your branch is rebased onto the latest main branch? I see a lot of changes that are already in our main branch (I'm not sure exactly why the show up here).

File: packages/app/.metadata
Comment (provokateurin) [reply packages/app/.metadata:1]: Looks like we forgot to move it from https://github.com/nextcloud/neon/blob/main/packages/neon/neon/.metadata

File: packages/app/ios/Runner/Assets.xcassets/AppIcon.appiconset/Icon-App-1024x1024@1x.png
Comment (nishp77) [reply packages/app/ios/Runner/Assets.xcassets/AppIcon.appiconset/Icon-App-1024x1024@1x.png:1]: So should just remove the icons? Where is the script?

File: packages/app/.metadata
Comment (nishp77) [reply packages/app/.metadata:1]: Should I delete the one under neon/neon?

File: packages/app/ios/Runner/Assets.xcassets/AppIcon.appiconset/Icon-App-1024x1024@1x.png
Comment (Leptopoda) [reply packages/app/ios/Runner/Assets.xcassets/AppIcon.appiconset/Icon-App-1024x1024@1x.png:1]: it's currently in `tool/generate-assets.sh`.

File: packages/app/.metadata
Comment (provokateurin) [reply packages/app/.metadata:1]: Better move the one from neon/neon and the start the create command again

**Comment by provokateurin** [comment]
@nishp77 do you plan on continuing this? Do you need some help with it? Afaik @Leptopoda has access to a Mac now, so you two could work on it together.

**Comment by Leptopoda** [comment]
I think the only thing missing was hooking it up to the icon generation script.
Other than that I tested this PR on an iOS emulator and it was working just fine.

Once I have more time I'll test it on some physical iOS devices and test all of our functionality but other than the mentioned notifications issue this should be all. 

When I checked out the PR I also noticed minor missing things like the package ID being wrong but that should also not be a big problem :)

**Comment by nishp77** [comment]
Sorry guys got busy because I was away I will starting working back again on this...

File: packages/app/.metadata
Comment (provokateurin) [inline packages/app/.metadata:1]: Merge conflicts :sweat_smile:

File: packages/app/macos/Flutter/ephemeral/flutter_export_environment.sh [hunk 3]
```diff
@@ -0,0 +1,12 @@
+#!/bin/sh
+# This is a generated file; do not edit or check into version control.
```
Comment (provokateurin) [inline packages/app/macos/Flutter/ephemeral/flutter_export_environment.sh:2]: :eyes:

**Comment by Leptopoda** [comment]
I'm going to close this for now and keep the branch locally.
It currently isn't our priority and I do not have the capacity to maintain iOS at this point of time.

**Pull request closed without merging**